target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-mr7500.dts

```diff
@@ -0,0 +1,648 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+// Copyright 2024 Weikai Kong (priv at pppig236.com)
+
+/dts-v1/;
+
+#include "ipq6018-512m.dtsi"
+#include "ipq6018-cp-cpu.dtsi"
+#include "ipq6018-ess.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/leds/common.h>
+
+/ {
+	model = "Linksys MR7500";
+	compatible = "linksys,mr7500", "qcom,ipq6018";
+	#address-cells = <0x2>;
+	#size-cells = <0x2>;
+	
+	aliases {
+		serial0 = &blsp1_uart3;
+		serial1 = &blsp1_uart2;
+		led-boot = &led_system_blue;
+		led-running = &led_system_blue;
+		led-failsafe = &led_system_red;
+		led-upgrade = &led_system_green;
+		ethernet0 = &dp1;
+		ethernet1 = &dp2;
+		ethernet2 = &dp3;
+		ethernet3 = &dp4;
+		ethernet4 = &dp5_syn;
+	};
+	
+	chosen {
+		stdout-path = "serial0:115200n8";
+		bootargs-append = " root=/dev/ubiblock0_0";
+	};
+	
+	extcon_usb: extcon_usb {
+		pinctrl-0 = <&extcon_usb_pins>;
+		pinctrl-names = "default";
+		id-gpio = <&tlmm 26 GPIO_ACTIVE_LOW>;
+		status = "okay";
+	};
+	
+	gpio_keys {
+		compatible = "gpio-keys";
+		pinctrl-0 = <&button_pins>;
+		pinctrl-names = "default";
+		
+		wps {
+			label = "wps";
+			linux,code = <KEY_WPS_BUTTON>;
+			gpios = <&tlmm 37 GPIO_ACTIVE_LOW>;
+		};
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&tlmm 56 GPIO_ACTIVE_LOW>;
+		};
+	};
+	
+	leds {
+		compatible = "pwm-leds";
+		usb {
+			label = "usb:system";
+			function = LED_FUNCTION_USB;
+			gpios = <&tlmm 30 GPIO_ACTIVE_HIGH>;
+			pwms = <&pwm 1 650000>;
+			max-brightness = <255>;
+			trigger-sources = <&usb3_port1>, <&usb3_port2>;
+			linux,default-trigger = "usbport";
+		};
+		
+		led_system_red: red {
+			label = "red:system";
+			pwms = <&pwm 2 1250000>;
+			max-brightness = <255>;
+		};
+		
+		led_system_green: green {
+			label = "green:system";
+			pwms = <&pwm 3 1250000>;
+			max-brightness = <255>;
+		};
+		
+		led_system_blue: blue {
+			label = "blue:system";
+			pwms = <&pwm 0 1250000>;
+			max-brightness = <255>;
+		};
+	};
+	
+	reg_usb_vbus: regulator-usb-vbus {
+		compatible = "regulator-fixed";
+		regulator-name = "usb_vbus";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		gpio = <&tlmm 25 GPIO_ACTIVE_LOW>;
+		regulator-always-on;
```

Comment:
Yeah, I doubt this is required, maybe `regulator-boot-on`

Reply:
I think I did this for usb to work, otherwise it is detected but immediately rejects the thumb drive for some reason

Reply:
nvm, it works now with only boot on